pkg/ingester/stream.go

```diff
@@ -234,7 +246,7 @@ func (s *stream) Push(
 		// Check if this this should be rate limited.
 		now := time.Now()
 		if !s.limiter.AllowN(now, s.tenant, len(entries[i].Line)) {
-			failedEntriesWithError = append(failedEntriesWithError, entryWithError{&entries[i], ErrStreamRateLimit})
+			failedEntriesWithError = append(failedEntriesWithError, entryWithError{&entries[i], &ErrStreamRateLimit{s.limiter.Limit(now, s.tenant), s.labelsString, len(entries[i].Line)}})
```

Comment:
I'd like to avoid calling this unnecessarily on the hotpath. Perhaps we can instead do this at the end of the batch.

Reply:
No sure I understand, this is exactly what we're doing for the other two cases where we append to `failedEntriesWithError` right below this line.